iib/workers/tasks/opm_operations.py

```diff
@@ -26,29 +31,194 @@
+
+    This stack of ports is used later in the Generator of the port numbers.
+
+    This function returns a tuple consisting of two elements:
+    1. List[List[int]]: Each inner list represents a set of port numbers used in a single attempt
+     of running the OPM command. Used port numbers are retrieved from the config iib_opm_port_ranges
+     values.
+    2. List[str]: Each string describes the intended purpose of the port number, with the position
+     of each string corresponding to the ports in the inner items in the first list.
+     This list is constructed from the config iib_opm_port_ranges keys.
+
+    Example:
+        ports, purposes = get_opm_port_stacks()
+        ports contains: [[50051, 50151], [50052, 50152]]
+        purposes content: ['opm_port', 'pprof_port']
+    In this example, ports 50051 and 50052 are port intended to use as 'opm_port' and ports 50151
+    and 50152 as 'pprof_port'.
+
+    :rtype: Tuple[List[List[int]], List[str]]
+    """
     conf = get_worker_config()
-    port_start = conf['iib_grpc_start_port']
-    port_end = port_start + conf['iib_grpc_max_port_tries']
+    port_purposes = list(conf.iib_opm_port_ranges)
 
-    port_stack = list(range(port_start, port_end))
-    random.shuffle(port_stack)
+    if parse_version(Opm.opm_version) < parse_version(conf.iib_opm_pprof_lock_required_min_version):
+        port_purposes.remove('opm_pprof_port')
 
-    log.debug('Get random ports from range [%d, %d)', port_start, port_end)
+    # get port_ranges we need for the give opm_version
+    port_ranges = [range(*conf.iib_opm_port_ranges[port_purpose]) for port_purpose in port_purposes]
 
-    while port_stack:
-        yield port_stack.pop(0)
+    ports_list = list(map(list, zip(*port_ranges)))
 
-    # The port stack is empty - we tried out all ports from allowed range
-    # therefore we will raise and IIB error
-    err_msg = f'No free port has been found after {port_end - port_start} attempts.'
-    log.error(err_msg)
-    raise IIBError(err_msg)
+    # shuffles the order, port pairs remain
+    random.shuffle(ports_list)
+
+    return ports_list, port_purposes
+
+
+def create_port_filelocks(func: Callable) -> Callable:
+    """
+    Create a file-lock on random port from the configured range.
+
+    :param function func: the function to be decorated    :return: the decorated function
+    :rtype: function
+    """
+
+    @wraps(func)
+    def inner(*args, **kwargs):
+
+        port_stacks, port_purposes = get_opm_port_stacks()
+
+        # shuffles the order, port pairs remain
+        random.shuffle(port_stacks)
+
+        # Attempt to acquire the lock for each port in the range (shuffled order)
+        lock_success = False
+
+        # Initialize the generator
+        gen = port_file_locks_generator(port_stacks=port_stacks, port_purposes=port_purposes)
+
+        # Use the function to retrieve values from the generator
+        while lock_success is not True:
```

Comment:
`while lock_success is not True`  
`while lock_success is False` ?
`while not lock_success` ?